packages/material-ui/src/Button/Button.js

```diff
@@ -157,6 +157,9 @@ const ButtonRoot = styled(ButtonBase, {
         backgroundColor: theme.palette.action.disabledBackground,
       }),
     },
+    [`&.${buttonClasses.inclusiveDisabled}`]: {
+      cursor: 'not-allowed',
+    },
```

Comment:
> I would propose we use the default cursor when disabled because we use the pointer cursor when enabled

This is already wrong. 

With your argumentation we could also use the drag cursor. You're missing why we should use the default curso.

Reply:
> You're missing why we should use the default cursor

The argumens for why was 1. It's how the native button behave 2. not-allowed looks "unfriendly".

Reply:
> 1. default cursor is how the native button behave

So you agree that we should also use `cursor: default` when the Button is not disabled because it's how the native button behaves?

>  2. not-allowed looks "unfriendly".

Don't agree there.

Reply:
@eps1lon I recall a similar discussion in https://github.com/mui-org/material-ui/issues/17871. I personally think that the expected behavior is button with `cursor: pointer` on the web, and `cursor: default` on the native UIs.

Why? It's how the apps I use on my OS, e.g. macOS Finder, and the websites e.g. GitHub I use on a day to day have trained my brain over the years. I would find something else confusing.
Is it the right behavior? Windows [explains](https://docs.microsoft.com/en-us/windows/win32/uxguide/inter-mouse?redirectedfrom=MSDN#hand-pointers) why on a native UI `cursor: default` should be used.

>  **Win32** / Design
> To avoid **confusion**, it is imperative not to use the hand pointer for other purposes

We see the same in https://developer.apple.com/design/human-interface-guidelines/macos/user-interaction/mouse-and-trackpad/ ("The content beneath the pointer is a URL link to a webpage, document, or other item.") and https://www.w3.org/TR/css-ui-3/#cursor ("The cursor is a pointer that indicates a link."). The community, at large, seems to have decided to override these guidelines. At the end of the day, I think that it's ultimately a convention to create predictability. So I think that going against the trend on the web would be wrong. It would be better to update these convention guidelines: https://github.com/w3c/csswg-drafts/issues/1936. Which they did: https://github.com/w3c/csswg-drafts/commit/272dd50ed23ee2fa24bc8a58d25378fc12711c73.

Back to argument 1. I think that it could work because we would use two different cursors between enabled and disabled.

---

> Don't agree there.

For 2. it's fair, I think that it's a matter of feeling, no objective arguments to find here :). I was personally influenced by https://github.com/twbs/bootstrap/issues/22222#issuecomment-292819959.

---

Since this is a UI/UX problem, in the end, I think that we can involve @danilo-leal, to bring a new perspective, as a designer.

Reply:
I actually don't have a new perspective to offer :sweat_smile: I read the CSS tricks article and learned from it, it makes a lot of sense. I also don't have any problem with the not-allowed cursor... Really liked how the experience turned out to be on this one. And happier that is making it more accessible :)

Reply:
This is a hot topic very polarized. From one side we have the "web guidelines" (from Microsoft, w3c, apple). On the other side, we have the actual user expectations.

Based on Apple website, expectations seem to be more important than "guidelines" for example, [the bag button](https://user-images.githubusercontent.com/14869087/130324370-84c7ef87-4c12-44f9-802b-c84fd38257ff.png), uses a pointer cursor.

My thoughts are better summarised by this [tweet from David K](https://twitter.com/DavidKPiano/status/1201552052784107521):
> So cursor: pointer is for navigation and not for performing some action.   
> You know who knows that rule?   
> Approximately 3% of web developers and 0.0000001% of users.

Matching user expectations is what makes me, personally, prefer to add a pointer cursor to buttons.

Then we have another point: **aesthetic tastes**. not-allowed look different from OS to OS [macOS example](https://user-images.githubusercontent.com/14869087/130324158-72ffd607-911a-4ead-8b1e-30b77d965a81.png). This might lead some people to override it in their apps because it's not aesthetically pleasing (a.k.a. ugly).

With all of that said... I don't have any strong preference, both sides have pros/cons. 

We already have the pointer cursor to distinguish clickable from non-clickable elements. If I had to make a decision, **I'd be inclined towards using `cursor: default` on disabled buttons** to keep the consistency with the current `disabled` prop that does not change the cursor either.

Reply:
> Matching user expectations is what makes me adding a pointer cursor to normal buttons.

I think web developers tunnel vision too much on the user interface they work on. Ordinary people don't know where the OS ends and the web begins. There's only a single "computer". The OS does not use `pointer` for buttons . There may be exceptions but the majority of clickable surfaces in macOS, Ubuntu (gnome) and Windows 10 do not use the `pointer` cursor.

> Then we have another point: aesthetic tastes. not-allowed cursor looks different from OS to OS macOS example. This might lead some people to override it in their apps because it's not aesthetically pleasing (a.k.a. ugly).

Doesn't that go against "matching expectations" i.e. "`not-allowed` cursor looks different from OS to OS macOS example." is exactly the point. People will recognize "not allowed" semantics since that's what they're used to from their computer.

> We already have the pointer cursor to distinguish clickable from non-clickable elements.

Not sure how this is relevant here. But it's controversial as well.

>  But if I had to take a side probably I'd be inclined towards using cursor: default on disabled buttons to keep the consistency with the current disabled prop at MUI that does not change the cursor either.

This is simply due to our current implementation. You currently cannot change the cursor for `<Button disabled />` since it's imlemented with `pointer-events: none`. So the argument for historical reasons doesn't apply here.